addons/binding/org.openhab.binding.lightify/src/main/java/org/openhab/binding/lightify/internal/discovery/LightifyDeviceDiscoveryService.java

```diff
@@ -0,0 +1,107 @@
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_ID;
+import static org.openhab.binding.lightify.internal.LightifyConstants.PROPERTY_ZONE_ID;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_BULB;
+import static org.openhab.binding.lightify.internal.LightifyConstants.THING_TYPE_LIGHTIFY_ZONE;
+import static org.openhab.binding.lightify.internal.LightifyUtils.exceptional;
+
+public class LightifyDeviceDiscoveryService extends AbstractDiscoveryService implements Consumer<LightifyLuminary> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LightifyDeviceDiscoveryService.class);
+
+    private final static int SEARCH_TIME = 60;
+
+    private final GatewayHandler gatewayHandler;
+
+    public LightifyDeviceDiscoveryService(GatewayHandler gatewayHandler) throws IllegalArgumentException {
+        super(DeviceHandler.SUPPORTED_TYPES, SEARCH_TIME, false);
+        this.gatewayHandler = gatewayHandler;
+    }
+
+    @Override
+    protected void startScan() {
+        exceptional(() -> {
+            LOGGER.info("Start scanning for paired devices");
+            LightifyLink lightifyLink = gatewayHandler.getLightifyLink();
+            lightifyLink.performSearch(this);
+        });
+    }
+
+    @Override
+    public void accept(LightifyLuminary luminary) {
+        exceptional(() -> {
+            LOGGER.info("Found device: " + luminary);
+            luminary.setSwitch(!luminary.isPowered(), null);
+            Thread.sleep(500);
+            luminary.setSwitch(!luminary.isPowered(), null);
+
+            DiscoveryResult discoveryResult = discoveryResult(luminary);
+            thingDiscovered(discoveryResult);
+        });
+    }
+
+    public void activate() {
+    }
+
+    public void deactivate() {
+    }
+
+    private DiscoveryResult discoveryResult(LightifyLuminary luminary) {
+        ThingTypeUID thingTypeUID = getThingTypeUID(luminary);
+        String deviceName = getDeviceName(luminary);
+
+        String deviceAddress = getDeviceAddress(luminary);
+        ThingUID bridgeUID = gatewayHandler.getThing().getUID();
+        ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID, deviceAddress);
+
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(PROPERTY_ID, thingUID.getId());
+        properties.put(PROPERTY_DEVICE_NAME, deviceName);
+        properties.put(PROPERTY_DEVICE_ADDRESS, luminary.address());
+
+        if (luminary instanceof LightifyZone) {
+            properties.put(PROPERTY_ZONE_ID, ((LightifyZone) luminary).getZoneId());
+        }
+
+        return DiscoveryResultBuilder.create(thingUID) //
+                                     .withBridge(bridgeUID) //
+                                     .withLabel(luminary.getName()) //
+                                     .withThingType(thingTypeUID) //
+                                     .withRepresentationProperty(PROPERTY_DEVICE_NAME) //
```

Comment:
Please remove `//` from the whole `return` statement and add javaDoc for the method